### PR TITLE
Initial commit of Server IAM submodule - 1

### PR DESCRIPTION
**Pull request by kevensen**
Continuing the sub-modularization work.  Will work on one component at a time.

I;ve tested this against the simple-example with:

1. `terraform apply`
2. On the server VM `forseti inventory create --import_as testmodel`
3. On the server VM `forseti model use testmodel`
4. On the server VM `forseti scanner run`
5. On the server VM `forseti notifier run`
6. `terraform destroy`

### CODE REVIEW

**Comment by kevensen** [comment]
Lint errors are due to "no space left on device".

**Comment by kevensen** [comment]
nm.  nothing to update in the README

File: main.tf [hunk 1]
```diff
@@ -254,6 +253,8 @@ module "server" {
   groups_settings_enabled                  = var.groups_settings_enabled
   groups_settings_violations_should_notify = var.groups_settings_violations_should_notify
 
+  server_iam_module = module.server_iam
```
Comment (hshin-g) [inline main.tf]: Can we get the formatting of this to match the other variables here?

File: modules/server_iam/variables.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  description = "The random suffix to append to all Forseti resources"
+}
+
+#-------------------------#
+# Forseti config notifier #
+#-------------------------#
+variable "cscc_violations_enabled" {
+  description = "Notify for CSCC violations"
+  default     = "false"
+}
```
Comment (hshin-g) [inline modules/server_iam/variables.tf]: Newline here

File: modules/server_iam/variables.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+variable "suffix" {
+  description = "The random suffix to append to all Forseti resources"
+}
+
+#-------------------------#
+# Forseti config notifier #
+#-------------------------#
+variable "cscc_violations_enabled" {
+  description = "Notify for CSCC violations"
+  default     = "false"
```
Comment (hshin-g) [inline modules/server_iam/variables.tf]: Add bool type and make string to bool

File: modules/server_iam/variables.tf [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  description = "GCP Folder that the Forseti project will be deployed into"
+  default     = ""
+}
+
+#-------#
+# Flags #
+#-------#
+variable "enable_write" {
+  description = "Enabling/Disabling write actions"
+  default     = "false"
```
Comment (hshin-g) [inline modules/server_iam/variables.tf]: Add bool type and make string to bool

File: modules/server_iam/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+output "forseti-server-service-account" {
+  description = "Forseti Server service account"
+  value       = google_service_account.forseti_server.email
+}
```
Comment (hshin-g) [inline modules/server_iam/outputs.tf]: Newline here

File: modules/server_iam/main.tf [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  folder = var.folder_id
+  member = "serviceAccount:${google_service_account.forseti_server.email}"
+}
+
+resource "google_organization_iam_member" "org_cscc" {
+  count  = var.org_id != "" && var.cscc_violations_enabled ? length(local.server_cscc_roles) : 0
+  role   = local.server_cscc_roles[count.index]
+  org_id = var.org_id
+  member = "serviceAccount:${google_service_account.forseti_server.email}"
+}
```
Comment (hshin-g) [inline modules/server_iam/main.tf]: Newline here

**Comment by kevensen** [comment]
Close in favor of #264

**Pull request closed without merging**